TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/kafka/KafkaTrafficCaptureSource.java

```diff
@@ -166,27 +172,30 @@ public CompletableFuture<List<ITrafficStreamWithKey>> readNextTrafficStreamChunk
         return CompletableFuture.supplyAsync(() -> {
             log.atTrace().setMessage("async...readNextTrafficStreamChunk()").log();
             return readNextTrafficStreamSynchronously();
-        });
+        }, kafkaExecutor);
     }
 
     public List<ITrafficStreamWithKey> readNextTrafficStreamSynchronously() {
         log.atTrace().setMessage("readNextTrafficStreamSynchronously()").log();
         try {
-            var records = trackingKafkaConsumer.getNextBatchOfRecords();
-            Stream<ITrafficStreamWithKey> trafficStream = StreamSupport.stream(records.spliterator(), false)
-                    .map(kafkaRecord -> {
+            return trackingKafkaConsumer.getNextBatchOfRecords((offsetData,kafkaRecord) -> {
                         try {
                             TrafficStream ts = TrafficStream.parseFrom(kafkaRecord.value());
+                            metricsLogger.atSuccess(MetricsEvent.PARSED_TRAFFIC_STREAM_FROM_KAFKA)
+                                    .setAttribute(MetricsAttributeKey.CONNECTION_ID, ts.getConnectionId())
+                                    .setAttribute(MetricsAttributeKey.TOPIC_NAME, trackingKafkaConsumer.topic)
+                                    .setAttribute(MetricsAttributeKey.SIZE_IN_BYTES, ts.getSerializedSize()).emit();
```

Comment:
Don't we already do this is in the next four lines?

Reply:
Shoot - yes we do.  Bad merge.  I've pulled one of these away